src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,156 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+
+public class Limelight extends SubsystemBase {
+    private NetworkTableInstance ntInstance = NetworkTableInstance.getDefault();
+    private NetworkTable table = ntInstance.getTable("/limelight");
+
+    private NetworkTableEntry yaw = table.getEntry("/tx");
+    private NetworkTableEntry pitch = table.getEntry("/ty");
+    private NetworkTableEntry lightMode = table.getEntry("/ledMode");
+    private NetworkTableEntry validTarget = table.getEntry("/tv");
+    private NetworkTableEntry skew = table.getEntry("/ts");
+    private NetworkTableEntry cameraMode = table.getEntry("/camMode");
+    private NetworkTableEntry pipeline = table.getEntry("/pipeline");
+    private NetworkTableEntry targetState = table.getEntry("/target_state");
+    private NetworkTableEntry poseData = table.getEntry("/camtran");
+
+    public static class Constants {
+        // LED MODES /limelight/ledMode
+        public static final int LED_MODE_FROM_PIPELINE = 0;
+        public static final int LED_MODE_FORCE_OFF = 1;
+        public static final int LED_MODE_FORCE_BLINK = 2;
+        public static final int LED_MODE_FORCE_ON = 3;
+
+        // pipeline numbers
+        public static final int PIPELINE_TELEOP = 0;
+        public static final int PIPELINE_GET_POS = 1;
+    }
+
+    // change with new robot; UNIT = inches
+    private static final double CAMERA_ELEVATION = 16.5;
+    private static final double TARGET_ELEVATION= 98.25;
+    // UNIT = degrees;
+    private static final double CAMERA_ANGLE = 13.5;
+
+    private Pose2d averagePose;
+
+    /**
+     * Constructor
+     */
+    public Limelight() {
+        yaw.setDefaultDouble(0);
+        pitch.setDefaultDouble(0);
+        lightMode.setDefaultNumber(0);
+        validTarget.setDefaultNumber(0);
+        skew.setDefaultDouble(0);
+        cameraMode.setDefaultNumber(0);
+        pipeline.setDefaultNumber(0);
+        targetState.setDefaultNumber(0);
+        poseData.setDefaultDoubleArray(new double[6]);
+    }
+
+    @Override
+    public void periodic() {
+        Pose2d currentPose = getPose();
+        if (currentPose != null)
+            averagePose = computeAveragePose(averagePose, currentPose);
+    }
+
+    public double getYaw() {
+        return this.yaw.getDouble(0);
+    }
+
+    public double getPitch() {
+        return pitch.getDouble(0);
+    }
+ 
+    public int getLightMode() {
+        return lightMode.getNumber(0).intValue();
+    }
+
+    public void setLightMode(int lightMode) {
+        this.lightMode.setNumber(lightMode);
+    }
+    
+    public boolean getValidTarget() {
```

Comment:
Minor: the name of this method strikes me as misleading. It implies that you are retrieving a target object of some sort, not checking whether a valid target exists. `getTargetValid`, `getTargetIsValid`, `isTargetValid` could all be better replacements.